openhtf/io/frontend/__main__.py

```diff
@@ -96,72 +123,36 @@ class StationStore(threading.Thread):
   """
   def __init__(self):
     super(StationStore, self).__init__()
-    self._death_event = threading.Event()
+    self._stop_event = threading.Event()
     self.hostname = socket.gethostname()
     self.stations = collections.defaultdict(Station)
 
-  @staticmethod
-  def MessageFramework(host, port, method='GET', message=None):
-    """Interact with a running instance of openhtf via its HTTP API.
-
-    The default method, 'GET', returns a JSON representation of station state,
-    while 'POST' can be used to respond to prompts. For example:
-
-      # Get the state of the openhtf instance at 192.168.2.25:10500.
-      station_state = MessageFramework(192.168.2.25, 10500)
-
-      # Respond to a prompt for the DUT's serial number:
-      MessageFramework(192.168.2.25, 10500, method='POST', 'SN-NCC1701D')
-
-    Args:
-      host: The IP address of the host running openhtf.
-      port: The TCP port of openhtf's HTTP API on host.
-      method: 'GET' or 'POST'.
-      message: Prompt response for POSTs. Ignored for GETs.
-    """
-    result = None
-    conn = httplib.HTTPConnection(host, int(port))
-    args = [method, '/']
-    if message and method == 'GET':
-      _LOG.warning("Message provided for 'GET' request will be ignored: %s",
-                   message)
-    if message and method == 'POST':
-      args.append(message)
-    try:
-      conn.request(*args)
-      if method == 'GET':
-        response = conn.getresponse()
-        if response.status == 200:
-          result = json.loads(response.read())
-    except socket.error as e:
-      result = None
-    conn.close()
-    return result
-
   def _Discover(self):
     """Use multicast to discover stations on the local network."""
     responses = multicast.send(PING_STRING,
                                FLAGS.discovery_address,
                                FLAGS.discovery_port,
                                FLAGS.discovery_ttl)
     for host, response in responses:
+      port = None
       try:
         port = json.loads(response)[PING_RESPONSE_KEY]
-        self._Track(host, port)
       except (KeyError, ValueError):
         _LOG.warning('Ignoring unrecognized discovery response from %s: %s' % (
             host, response))
+      self._Track(host, port)
```

Comment:
hm, moving this outside the try block means you might try to `_Track(host, None)`, but that's going to fail because there's no port set.  is that intended?